src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,564 @@
+
+            stream.WriteInt32(Algorithm.Code);
+            stream.WriteLength(Digest.Length);
+            stream.WriteSomeBytes(Digest);
+        }
+
+        private void Read(Stream stream)
+        {
+            using (var cis = new CodedInputStream(stream, true))
+            {
+                Read(cis);
+            }
+        }
+
+        private void Read(CodedInputStream stream)
+        {
+            var code = stream.ReadInt32();
+            var digestSize = stream.ReadLength();
+
+            HashingAlgorithm.Codes.TryGetValue(code, out var a);
+            Algorithm = a;
+            if (Algorithm == null)
+            {
+                Algorithm = HashingAlgorithm.Register("ipfs-" + code, code, digestSize);
+                RaiseUnknownHashingAlgorithm(Algorithm);
+            }
+            else if (Algorithm.DigestSize != 0 && digestSize != Algorithm.DigestSize)
+            {
+                throw new InvalidDataException(string.Format("The digest size {0} is wrong for {1}; it should be {2}.",
+                    digestSize, Algorithm.Name, Algorithm.DigestSize));
+            }
+
+            Digest = stream.ReadSomeBytes(digestSize);
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode() { return ToString().GetHashCode(); }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            var that = obj as MultiHash;
+            return that == null
+                ? false
+                : Equals(that);
+        }
+
+        /// <inheritdoc />
+        public bool Equals(MultiHash that)
+        {
+            return Algorithm.Code == that.Algorithm.Code
+             && Digest.SequenceEqual(that.Digest);
+        }
+
+        /// <summary>
+        ///   Value equality.
+        /// </summary>
+        public static bool operator ==(MultiHash a, MultiHash b)
+        {
+            if (ReferenceEquals(a, b)) return true;
+            if (a is null) return false;
+            if (b is null) return false;
+
+            return a.Equals(b);
+        }
+
+        /// <summary>
+        ///   Value inequality.
+        /// </summary>
+        public static bool operator !=(MultiHash a, MultiHash b) { return !(a == b); }
+
+        /// <summary>
+        ///   Returns the <see cref="Base58"/> encoding of the <see cref="MultiHash"/>.
+        /// </summary>
+        /// <returns>
+        ///   A base-58 representaton of the MultiHash.
+        /// </returns>
+        /// <seealso cref="ToBase58"/>
+        public override string ToString() { return ToBase58(); }
+
+        /// <summary>
+        ///   Returns the <see cref="Base58"/> encoding of the <see cref="MultiHash"/>.
+        /// </summary>
+        /// <returns>
+        ///   The <see cref="Base58"/> representation of the <see cref="MultiHash"/>.
+        /// </returns>
+        public string ToBase58()
+        {
+            if (_b58String != null) return _b58String;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)